datafusion/functions-nested/src/set_ops.rs

```diff
@@ -290,10 +290,14 @@ impl ScalarUDFImpl for ArrayDistinct {
 
     fn return_type(&self, arg_types: &[DataType]) -> Result<DataType> {
         match &arg_types[0] {
-            List(field) => Ok(DataType::new_list(field.data_type().clone(), true)),
-            LargeList(field) => {
-                Ok(DataType::new_large_list(field.data_type().clone(), true))
-            }
+            List(field) => Ok(DataType::new_list(
+                field.data_type().clone(),
+                field.is_nullable(),
+            )),
+            LargeList(field) => Ok(DataType::new_large_list(
+                field.data_type().clone(),
+                field.is_nullable(),
+            )),
```

Comment:
I looked at a similar PR for `map_values`: #15901,
but I wasn't sure should I switch to `return_field_from_args`

Reply:
Thank you, I applied the suggestion.